frame/treasury/src/lib.rs

```diff
@@ -393,6 +395,36 @@ pub mod pallet {
 				.map_err(|_| Error::<T, I>::TooManyApprovals)?;
 			Ok(())
 		}
+
+		/// Force a previously approved proposal to be removed from the approval queue. 
+		///
+		/// May only be called from `T::RejectOrigin`.
+		///
+		/// # <weight>
+		/// - Complexity: O(A) where `A` is the number of approvals
+		/// - DbReads: `Proposals`, `Approvals`
+		/// - DbWrites: `Approvals`
+		/// # </weight>
+		#[pallet::weight((T::WeightInfo::remove_approval(), DispatchClass::Operational))]
+		pub fn remove_approval(
+			origin: OriginFor<T>,
+			#[pallet::compact] proposal_id: ProposalIndex,
+		) -> DispatchResult {
+			T::RejectOrigin::ensure_origin(origin)?;
+
+			ensure!(<Proposals<T, I>>::contains_key(proposal_id), Error::<T, I>::InvalidIndex);
```

Comment:
Doesn't seem to me like we really need this check?

If it exists in the `Approvals`, that is all we want to do to remove it.

Reply:
Resolved